packages/flutter/test/material/scaffold_test.dart

```diff
@@ -2800,6 +2800,43 @@ void main() {
     // The scrim should be gone.
     expect(findModalBarrier(), findsNothing);
   });
+
+  testWidgets('Animated closing bottom sheet and removing floating action button', (WidgetTester tester) async {
+    PersistentBottomSheetController<void>? controller;
+    bool show = true;
+    late StateSetter setState;
+
+    await tester.pumpWidget(StatefulBuilder(builder: (_, StateSetter stateSetter) {
+      setState = stateSetter;
+      return MaterialApp(
+        home: Scaffold(
+          body: Builder(builder: (BuildContext context) {
+            WidgetsBinding.instance.addPostFrameCallback((_) {
```

Comment:
Is `addPostFrameCallback` necessary? 
It'd be preferable to add and remove the bottom sheet by pressing buttons.

Reply:
Thanks for the comments. I rewrite the test using buttons.

> Yeah, maybe can the FAB.onPressed trigger this change?

I also tried to trigger 2 changes at FAB.onPressed at once, but it does not reproduce the assertion error.

While rewriting the test, I found the problem is only that while the hiding FAB animation is running, another event starts the closing bottom animation. The reversed order (close bottom sheet → hide FAB) or both at once does not.